src/main/java/com/google/sps/servlets/TripServlet.java

```diff
@@ -80,40 +97,64 @@
   private static final String START_TIME = "start-time";
   private static final String TRAVEL_TIME = "travel-time";
 
+  // Constants to get form inputs.
+  private static final String INPUT_TRIP_NAME = "inputTripName";
+  private static final String INPUT_DESTINATION = "inputDestination";
+  private static final String INPUT_DAY_OF_TRAVEL = "inputDayOfTravel";
+  private static final String INPUT_POI_LIST = "poiList";
+
+  // Datastore and API context
+  private DatastoreService datastore;
+  private GeoApiContext context;
+
+  /**
+   * Initializes datastore and API.
+   */
   @Override
   public void init() {
+    this.datastore = DatastoreServiceFactory.getDatastoreService();  
     this.context = new GeoApiContext.Builder()
       .apiKey(Config.API_KEY)
       .build();
   }
 
+  /**
+   * Get user input.
+   * Generate directionsRequest from user input and parse optimized route.
+   * Create and store events in Datastore.
+   */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) 
       throws IOException {
     response.setContentType("application/json;");
 
-    // global info needed
-    DatastoreService datastore = 
-                                DatastoreServiceFactory.getDatastoreService();
-    Enumeration<String> params = request.getParameterNames(); 
-
     // Retrieve form inputs to define the Trip object.
     this.tripName = request.getParameter(INPUT_TRIP_NAME);
     this.tripDestination = request.getParameter(INPUT_DESTINATION);
     this.tripDayOfTravel = request.getParameter(INPUT_DAY_OF_TRAVEL);
+    String[] poiStrings = request.getParameterValues(INPUT_POI_LIST);
 
     // Populate the destinationName and photoSrc fields using Google Maps API.
-    populateDestinationAndPhoto(context, tripDestination);
+    populateDestinationAndPhoto(this.context, this.tripDestination);
 
     // Store the Trip Entity in datastore with the User Entity as an ancestor.
     Entity tripEntity = storeTripEntity(response, this.tripName, this.destinationName, 
       this.tripDayOfTravel, this.photoSrc, datastore);
 
+    //do post for maps
+    DirectionsApiRequest dirRequest = generateDirectionsRequest(this.tripDestination, this.tripDestination, poiStrings, this.context);
+    DirectionsResult dirResult = getDirectionsResult(dirRequest);
+    List<Integer> travelTimes = getTravelTimes(dirResult);
+    List<String> orderedLocationStrings = getOrderedWaypoints(dirResult, poiStrings);
+
     // put TripDay entity into datastore
-    Entity tripDayEntity = putTripDayInDatastore(request, datastore, LocalDate.parse(tripDayOfTravel), tripEntity.getKey());
+    Entity tripDayEntity = putTripDayInDatastore(this.tripDestination, datastore, LocalDate.parse(tripDayOfTravel), tripEntity.getKey());
 
+    List<Entity> locationEntities = TripDay.locationsToEntities(orderedLocationStrings, tripDayEntity.getKey());
+    TripDay.storeLocationsInDatastore(locationEntities, this.datastore);
+    
```

Comment:
No change required.

These locationEntities seem a little redundant with the Event entities added below, since it seems like for each POI there's one locationEntity and one Event. I'm wondering if it'd be possible to just add an 'order' field to the Event entities then do away with the locationEntities? Either way no change needed for this CL.

Reply:
Yes, that's a good point! Can definitely working on consolidating these post MVP